### PR TITLE
Store.hasRecordForId should true only if a record for a given type and ID is already loaded.

### PR DESCRIPTION
**Pull request by bmac**
This pr is built on top of #2659

### CODE REVIEW

File: packages/ember-data/tests/unit/store/has_record_for_id_test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    store = env.store;
+
+  },
+
+  teardown: function() {
+    Ember.run(store, 'destroy');
+  }
+});
+
+test("hasRecordForId should return false for records in the empty state ", function() {
```
Comment (igorT) [inline packages/ember-data/tests/unit/store/has_record_for_id_test.js]: How about a test for true while we are at it?

Comment (bmac) [inline packages/ember-data/tests/unit/store/has_record_for_id_test.js]: Will do

Comment (bmac) [inline packages/ember-data/tests/unit/store/has_record_for_id_test.js]: Added a test for for the true case.

File: packages/ember-data/tests/unit/store/has_record_for_id_test.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      lastName: "Katz",
+      phoneNumbers: [1]
+    });
+
+    equal(false, store.hasRecordForId(PhoneNumber, 1), 'hasRecordForId only returns true for loaded records');
+
+  });
+});
+
+test("hasRecordForId should return false for records in the loaded state ", function() {
```
Comment (sly7-7) [inline packages/ember-data/tests/unit/store/has_record_for_id_test.js]: I think you meant `true` here :hamster:

Comment (bmac) [inline packages/ember-data/tests/unit/store/has_record_for_id_test.js]: Good catch :dancers:

**Pull request merged**